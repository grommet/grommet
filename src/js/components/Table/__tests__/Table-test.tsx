import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import {
  Grommet,
  Table,
  TableHeader,
  TableFooter,
  TableBody,
  TableRow,
  TableCell,
} from '../..';

test('Table renders', () => {
  const { container } = render(
    <Grommet>
      <Table />
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('Table renders outside grommet', () => {
  const { container } = render(<Table />);

  expect(container.firstChild).toMatchSnapshot();
});

test('Table caption renders', () => {
  const { container } = render(
    <Grommet>
      <Table caption="Caption" />
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('TableHeader renders', () => {
  const { container } = render(
    <Grommet>
      <Table>
        <TableHeader />
      </Table>
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('TableFooter renders', () => {
  const { container } = render(
    <Grommet>
      <Table>
        <TableFooter />
      </Table>
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('TableBody renders', () => {
  const { container } = render(
    <Grommet>
      <Table>
        <TableBody />
      </Table>
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('TableRow renders', () => {
  const { container } = render(
    <Grommet>
      <Table>
        <TableBody>
          <TableRow />
        </TableBody>
      </Table>
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('TableCell renders', () => {
  const { container } = render(
    <Grommet>
      <Table>
        <TableHeader>
          <TableRow>
            <TableCell />
          </TableRow>
        </TableHeader>
        <TableBody>
          <TableRow>
            <TableCell />
          </TableRow>
        </TableBody>
        <TableFooter>
          <TableRow>
            <TableCell />
          </TableRow>
        </TableFooter>
      </Table>
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('TableCell scope renders', () => {
  const { container } = render(
    <Grommet>
      <Table>
        <TableHeader>
          <TableRow>
            <TableCell scope="col" />
          </TableRow>
        </TableHeader>
        <TableBody>
          <TableRow>
            <TableCell scope="row" />
          </TableRow>
        </TableBody>
      </Table>
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('TableCell size renders', () => {
  const { container } = render(
    <Grommet>
      <Table>
        <TableBody>
          <TableRow>
            <TableCell size="xsmall" />
            <TableCell size="small" />
            <TableCell size="medium" />
            <TableCell size="large" />
          </TableRow>
        </TableBody>
      </Table>

      <Table>
        <TableBody>
          <TableRow>
            <TableCell size="1/2" />
            <TableCell size="2/4" />
          </TableRow>
        </TableBody>
      </Table>

      <Table>
        <TableBody>
          <TableRow>
            <TableCell size="1/3" />
            <TableCell size="2/3" />
          </TableRow>
        </TableBody>
      </Table>

      <Table>
        <TableBody>
          <TableRow>
            <TableCell size="1/4" />
            <TableCell size="3/4" />
          </TableRow>
        </TableBody>
      </Table>
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('TableCell verticalAlign renders', () => {
  const { container } = render(
    <Grommet>
      <Table>
        <TableHeader>
          <TableRow>
            <TableCell verticalAlign="top" />
            <TableCell verticalAlign="middle" />
            <TableCell verticalAlign="bottom" />
          </TableRow>
        </TableHeader>
      </Table>
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('TableCell plain renders', () => {
  const { container } = render(
    <Grommet>
      <Table>
        <TableHeader>
          <TableRow>
            <TableCell plain />
          </TableRow>
        </TableHeader>
      </Table>
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('TableCell border renders', () => {
  const { container } = render(
    <Grommet>
      <Table>
        <TableHeader>
          <TableRow>
            <TableCell border="top" />
            <TableCell border={{ side: 'top', size: 'medium' }} />
            <TableCell
              border={[
                { side: 'right', size: 'medium' },
                { side: 'bottom', size: 'large' },
              ]}
            />
          </TableRow>
        </TableHeader>
      </Table>
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('Table with ref', () => {
  const ref = React.createRef<HTMLTableElement>();
  render(
    <Grommet>
      <Table ref={ref} />
    </Grommet>,
  );

  expect(ref.current).not.toBeNull();
});

test('theme caption margin', () => {
  const customTheme = {
    table: {
      caption: {
        margin: 'large',
      },
    },
  };

  const { container } = render(
    <Grommet theme={customTheme}>
      <Table caption="Test Caption" />
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});
