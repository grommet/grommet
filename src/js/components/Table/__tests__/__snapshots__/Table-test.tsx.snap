// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Table caption renders 1`] = `
.c2 {
  margin-bottom: 3px;
}

.c1 {
  border-spacing: 0;
  border-collapse: collapse;
  width: inherit;
}

.c1 tbody tr:last-child td {
  border-color: transparent;
}

.c1 tbody tr:last-child th {
  border-color: transparent;
}

.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  <table
    class="c1"
  >
    <caption
      class="c2"
    >
      Caption
    </caption>
  </table>
</div>
`;

exports[`Table renders 1`] = `
.c1 {
  border-spacing: 0;
  border-collapse: collapse;
  width: inherit;
}

.c1 tbody tr:last-child td {
  border-color: transparent;
}

.c1 tbody tr:last-child th {
  border-color: transparent;
}

.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  <table
    class="c1"
  />
</div>
`;

exports[`Table renders outside grommet 1`] = `
.c0 {
  border-spacing: 0;
  border-collapse: collapse;
  width: inherit;
}

<table
  class="c0"
/>
`;

exports[`TableBody renders 1`] = `
.c1 {
  border-spacing: 0;
  border-collapse: collapse;
  width: inherit;
}

.c1 tbody tr:last-child td {
  border-color: transparent;
}

.c1 tbody tr:last-child th {
  border-color: transparent;
}

.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  <table
    class="c1"
  >
    <tbody
      class="StyledTable__StyledTableBody-sc-1m3u5g-3"
    />
  </table>
</div>
`;

exports[`TableCell border renders 1`] = `
.c2 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  text-align: start;
  border-top: solid 1px rgba(0, 0, 0, 0.33);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
}

.c3 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  text-align: start;
  border-top: solid 4px rgba(0, 0, 0, 0.33);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
}

.c4 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  text-align: start;
  border-right: solid 4px rgba(0, 0, 0, 0.33);
  border-bottom: solid 12px rgba(0, 0, 0, 0.33);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
}

.c1 {
  border-spacing: 0;
  border-collapse: collapse;
  width: inherit;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  <table
    class="c1"
  >
    <thead
      class="StyledTable__StyledTableHeader-sc-1m3u5g-4"
    >
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2"
      >
        <td
          class="c2"
        />
        <td
          class="c3"
        />
        <td
          class="c4"
        />
      </tr>
    </thead>
  </table>
</div>
`;

exports[`TableCell plain renders 1`] = `
.c2 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  text-align: start;
  border-bottom: solid 1px rgba(0, 0, 0, 0.36);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-top: 12px;
  padding-bottom: 11px;
}

.c2 >div {
  height: 100%;
  justify-content: center;
}

.c2 label {
  justify-content: center;
}

.c1 {
  border-spacing: 0;
  border-collapse: collapse;
  width: inherit;
}

.c1 tbody tr:last-child td {
  border-color: transparent;
}

.c1 tbody tr:last-child th {
  border-color: transparent;
}

.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  <table
    class="c1"
  >
    <thead
      class="StyledTable__StyledTableHeader-sc-1m3u5g-4"
    >
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2"
      >
        <td
          class="c2"
        />
      </tr>
    </thead>
  </table>
</div>
`;

exports[`TableCell renders 1`] = `
.c3 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c2 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  text-align: start;
  border-bottom: solid 1px rgba(0, 0, 0, 0.36);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-top: 12px;
  padding-bottom: 11px;
}

.c2 >div {
  height: 100%;
  justify-content: center;
}

.c2 label {
  justify-content: center;
}

.c4 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  text-align: start;
  border-bottom: solid 1px rgba(0, 0, 0, 0.12);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-top: 12px;
  padding-bottom: 11px;
}

.c5 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  text-align: start;
  border-top: solid 1px rgba(0, 0, 0, 0.36);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-top: 11px;
  padding-bottom: 12px;
  font-weight: undefined;
}

.c1 {
  border-spacing: 0;
  border-collapse: collapse;
  width: inherit;
}

.c1 tbody tr:last-child td {
  border-color: transparent;
}

.c1 tbody tr:last-child th {
  border-color: transparent;
}

.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  <table
    class="c1"
  >
    <thead
      class="StyledTable__StyledTableHeader-sc-1m3u5g-4"
    >
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2"
      >
        <td
          class="c2"
        >
          <div
            class="c3"
          />
        </td>
      </tr>
    </thead>
    <tbody
      class="StyledTable__StyledTableBody-sc-1m3u5g-3"
    >
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2"
      >
        <td
          class="c4"
        />
      </tr>
    </tbody>
    <tfoot
      class="StyledTable__StyledTableFooter-sc-1m3u5g-5"
    >
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2"
      >
        <td
          class="c5"
        >
          <div
            class="c3"
          />
        </td>
      </tr>
    </tfoot>
  </table>
</div>
`;

exports[`TableCell scope renders 1`] = `
.c3 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c2 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  text-align: start;
  border-bottom: solid 1px rgba(0, 0, 0, 0.36);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-top: 12px;
  padding-bottom: 11px;
}

.c2 >div {
  height: 100%;
  justify-content: center;
}

.c2 label {
  justify-content: center;
}

.c4 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  text-align: start;
  border-bottom: solid 1px rgba(0, 0, 0, 0.12);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-top: 12px;
  padding-bottom: 11px;
}

.c1 {
  border-spacing: 0;
  border-collapse: collapse;
  width: inherit;
}

.c1 tbody tr:last-child td {
  border-color: transparent;
}

.c1 tbody tr:last-child th {
  border-color: transparent;
}

.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  <table
    class="c1"
  >
    <thead
      class="StyledTable__StyledTableHeader-sc-1m3u5g-4"
    >
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2"
      >
        <th
          class="c2"
          scope="col"
        >
          <div
            class="c3"
          />
        </th>
      </tr>
    </thead>
    <tbody
      class="StyledTable__StyledTableBody-sc-1m3u5g-3"
    >
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2"
      >
        <th
          class="c4"
          scope="row"
        />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`TableCell size renders 1`] = `
.c2 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  width: 96px;
  max-width: 96px;
  overflow: hidden;
  text-align: start;
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
}

.c3 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  width: 192px;
  max-width: 192px;
  overflow: hidden;
  text-align: start;
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
}

.c4 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  width: 384px;
  max-width: 384px;
  overflow: hidden;
  text-align: start;
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
}

.c5 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  width: 768px;
  max-width: 768px;
  overflow: hidden;
  text-align: start;
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
}

.c6 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  width: 50%;
  max-width: 50%;
  overflow: hidden;
  text-align: start;
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
}

.c7 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  width: 33.33%;
  max-width: 33.33%;
  overflow: hidden;
  text-align: start;
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
}

.c8 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  width: 66.66%;
  max-width: 66.66%;
  overflow: hidden;
  text-align: start;
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
}

.c9 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  width: 25%;
  max-width: 25%;
  overflow: hidden;
  text-align: start;
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
}

.c10 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  width: 75%;
  max-width: 75%;
  overflow: hidden;
  text-align: start;
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
}

.c1 {
  border-spacing: 0;
  border-collapse: collapse;
  width: inherit;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  <table
    class="c1"
  >
    <tbody
      class="StyledTable__StyledTableBody-sc-1m3u5g-3"
    >
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2"
      >
        <td
          class="c2"
        />
        <td
          class="c3"
        />
        <td
          class="c4"
        />
        <td
          class="c5"
        />
      </tr>
    </tbody>
  </table>
  <table
    class="c1"
  >
    <tbody
      class="StyledTable__StyledTableBody-sc-1m3u5g-3"
    >
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2"
      >
        <td
          class="c6"
        />
        <td
          class="c6"
        />
      </tr>
    </tbody>
  </table>
  <table
    class="c1"
  >
    <tbody
      class="StyledTable__StyledTableBody-sc-1m3u5g-3"
    >
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2"
      >
        <td
          class="c7"
        />
        <td
          class="c8"
        />
      </tr>
    </tbody>
  </table>
  <table
    class="c1"
  >
    <tbody
      class="StyledTable__StyledTableBody-sc-1m3u5g-3"
    >
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2"
      >
        <td
          class="c9"
        />
        <td
          class="c10"
        />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`TableCell verticalAlign renders 1`] = `
.c3 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  justify-content: flex-start;
}

.c5 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  justify-content: center;
}

.c7 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  justify-content: flex-end;
}

.c2 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  vertical-align: top;
  text-align: start;
  border-bottom: solid 1px rgba(0, 0, 0, 0.36);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-top: 12px;
  padding-bottom: 11px;
}

.c2 >div {
  height: 100%;
  justify-content: center;
}

.c2 label {
  justify-content: center;
}

.c4 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  vertical-align: middle;
  text-align: start;
  border-bottom: solid 1px rgba(0, 0, 0, 0.36);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-top: 12px;
  padding-bottom: 11px;
}

.c4 >div {
  height: 100%;
  justify-content: center;
}

.c4 label {
  justify-content: center;
}

.c6 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  vertical-align: bottom;
  text-align: start;
  border-bottom: solid 1px rgba(0, 0, 0, 0.36);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-top: 12px;
  padding-bottom: 11px;
}

.c6 >div {
  height: 100%;
  justify-content: center;
}

.c6 label {
  justify-content: center;
}

.c1 {
  border-spacing: 0;
  border-collapse: collapse;
  width: inherit;
}

.c1 tbody tr:last-child td {
  border-color: transparent;
}

.c1 tbody tr:last-child th {
  border-color: transparent;
}

.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  <table
    class="c1"
  >
    <thead
      class="StyledTable__StyledTableHeader-sc-1m3u5g-4"
    >
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2"
      >
        <td
          class="c2"
        >
          <div
            class="c3"
          />
        </td>
        <td
          class="c4"
        >
          <div
            class="c5"
          />
        </td>
        <td
          class="c6"
        >
          <div
            class="c7"
          />
        </td>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`TableFooter renders 1`] = `
.c1 {
  border-spacing: 0;
  border-collapse: collapse;
  width: inherit;
}

.c1 tbody tr:last-child td {
  border-color: transparent;
}

.c1 tbody tr:last-child th {
  border-color: transparent;
}

.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  <table
    class="c1"
  >
    <tfoot
      class="StyledTable__StyledTableFooter-sc-1m3u5g-5"
    />
  </table>
</div>
`;

exports[`TableHeader renders 1`] = `
.c1 {
  border-spacing: 0;
  border-collapse: collapse;
  width: inherit;
}

.c1 tbody tr:last-child td {
  border-color: transparent;
}

.c1 tbody tr:last-child th {
  border-color: transparent;
}

.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  <table
    class="c1"
  >
    <thead
      class="StyledTable__StyledTableHeader-sc-1m3u5g-4"
    />
  </table>
</div>
`;

exports[`TableRow renders 1`] = `
.c1 {
  border-spacing: 0;
  border-collapse: collapse;
  width: inherit;
}

.c1 tbody tr:last-child td {
  border-color: transparent;
}

.c1 tbody tr:last-child th {
  border-color: transparent;
}

.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  <table
    class="c1"
  >
    <tbody
      class="StyledTable__StyledTableBody-sc-1m3u5g-3"
    >
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2"
      />
    </tbody>
  </table>
</div>
`;
